### ISSUE TITLE
implement something like "accessory.unpublish()" or "accessory.isPublished()"

### ISSUE THREAD

**Issue by oliverrahner** [KhaosT/HAP-NodeJS#579]
Hi,

I recently forked the node-red implementation of HAP-NodeJS and ran across an issue.
The current implementation was buggy when redeploying node-red because node state is supposed to be "reset" after a new deploy.

Unfortunately, there is no way to either determine if an accessory is currently being published or to "un-publish" it.

This leads to the problem that my new implementation tries to re-publish a service after redeploying a node-red flow which then crashes node-red:

```
16 Apr 21:18:15 - [red] Uncaught Exception:
16 Apr 21:18:15 - Error: dns service error: name conflict
    at Error (native)
    at Advertisement.on_service_registered (/usr/lib/node_modules/@oliverrahner/node-red-contrib-homekit-bridged/node_modules/mdns/lib/advertisement.js:42:24)
    at SocketWatcher.MDNSService.self.watcher.callback (/usr/lib/node_modules/@oliverrahner/node-red-contrib-homekit-bridged/node_modules/mdns/lib/mdns_service.js:18:40)
```

EDIT:
Just tried a 'proof-of-concept' using the option to un-publish the server.
This works:

```
   this.on('close', function (removed, done) {
      if (removed) {
        // This node has been deleted
        bridge.destroy()
      } else {
        // This node is being restarted
        bridge._server.stop()
        bridge._server = undefined
        bridge._advertiser.stopAdvertising()
        bridge._advertiser = undefined
        bridge = null
      }
      done()
    })
```

After that, the bridge seems to be properly cleaned up so that my re-publishing works.

**Comment by Tkg8455** [comment]
Hello,
Can someone help me to understand how to use "Characteristic" issues. I am not familiar with programming.
Can you show a way to understand?

Thank you very much

**Comment by oliverrahner** [comment]
Closed in reference to #580
